ironfish/src/network/peers/addressManager.ts

```diff
@@ -2,24 +2,149 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
 import { HostsStore } from '../../fileStores'
+import { FileSystem } from '../../fileSystems'
+import { ArrayUtils } from '../../utils'
+import { Peer, PeerList } from '..'
+import { ConnectionDirection, ConnectionType } from './connections'
 import { PeerAddress } from './peerAddress'
 
 /**
  * AddressManager stores the necessary data for connecting to new peers
  * and provides functionality for persistence of said data.
  */
 export class AddressManager {
-  hosts: HostsStore
+  hostsStore: HostsStore
 
-  constructor(hostsStore: HostsStore) {
-    this.hosts = hostsStore
+  constructor(files: FileSystem, dataDir?: string) {
+    this.hostsStore = new HostsStore(files, dataDir)
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    this.hostsStore.load()
   }
 
   get priorConnectedPeerAddresses(): ReadonlyArray<Readonly<PeerAddress>> {
-    return this.hosts.getArray('priorPeers')
+    return this.hostsStore.getArray('priorPeers')
   }
 
   get possiblePeerAddresses(): ReadonlyArray<Readonly<PeerAddress>> {
-    return this.hosts.getArray('possiblePeers')
+    return this.hostsStore.getArray('possiblePeers')
+  }
+  /**
+   * Adds addresses associated to peers received from peer list
+   */
+  addAddressesFromPeerList(peerList: PeerList): void {
+    const newAddresses: PeerAddress[] = peerList.payload.connectedPeers.map((peer) => ({
+      address: peer.address,
+      port: peer.port,
+      identity: peer.identity ?? null,
```

Comment:
I don't think identity can be nullable in the peer list -- we should update the types if it is